### PR TITLE
Restore desired soft deprecation behaviour

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/deprecated.R [hunk 1]
```diff
@@ -102,14 +102,15 @@ deprecate_soft <- function(when,
...
-  } else if (verbosity %in% "warning" ||
-             (is_string(verbosity, "default") && env_inherits_global(user_env))) {
-    trace <- trace_back(bottom = caller_env())
-    deprecate_warn0(msg, trace, always = TRUE)
+  } else if (verbosity %in% c("warning", "default")) {
+    if (env_inherits_global(user_env) || from_testthat(user_env)) {
+      trace <- trace_back(bottom = caller_env())
+      deprecate_warn0(msg, trace, always = verbosity == "warning")
+    } else {
+      deprecate_soft0(msg)
```
Comment (lionel-) [inline R/deprecated.R:110]: Inline `deprecate_soft0()` or rename to `signal_soft_deprecation()`? The name confused me for a while.

**Review by lionel- — APPROVED** [review]
Looks like a regression from https://github.com/r-lib/lifecycle/pull/73.

File: R/deprecated.R (hunk 1 above)
Comment (hadley) [reply R/deprecated.R]: I refactored both `deprecate_soft()` and `deprecate_warn()` to make the logic more clear.

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # lifecycle (development version)
 
+* `deprecate_soft()` now only deprecates in tests if you are testing
+  locally (not on CRAN) and the function comes from the current package.
```
Comment (DavisVaughan) [inline NEWS.md:4]:
```suggestion
* `deprecate_soft()` now only warns in tests if you are testing
  locally (not on CRAN) and the deprecated function is called
  directly in the package being tested.
```

I found the current explanation rather confusing

Also, does it warn on CI?

**Review by DavisVaughan — COMMENTED** [review]
Pausing review to wait for changes

**Pull request merged**